netty/src/main/java/io/grpc/netty/GrpcSslContexts.java

```diff
@@ -172,9 +185,22 @@ static void ensureAlpnAndH2Enabled(ApplicationProtocolNegotiator alpnNegotiator)
     checkArgument(alpnNegotiator != null, "ALPN must be configured");
     checkArgument(alpnNegotiator.protocols() != null && !alpnNegotiator.protocols().isEmpty(),
         "ALPN must be enabled and list HTTP/2 as a supported protocol.");
-    if (!alpnNegotiator.protocols().contains(HTTP2_VERSION)) {
-      throw new IllegalArgumentException("This ALPN config does not support HTTP/2. Expected '"
-          + HTTP2_VERSION + "', but got " + alpnNegotiator.protocols() + '.');
-    }
+    checkArgument(
+        alpnNegotiator.protocols().contains(HTTP2_VERSION),
+        "This ALPN config does not support HTTP/2. Expected %s, but got %s'.",
+        HTTP2_VERSION,
+        alpnNegotiator.protocols());
+    checkArgument(
+        alpnNegotiator.protocols().contains(GRPC_EXP_VERSION),
+        "This ALPN config does not support gRPC upgrade. Expected %s, but got %s'.",
+        GRPC_EXP_VERSION,
+        alpnNegotiator.protocols());
+    checkArgument(
```

Comment:
I've removed the check. The original rationale was that it would be not all that useful (and likely indicate a mistaken use of grpc-ext) to preference plain h2 over it, since all gRPC servers should support h2 and will as a result never make use of grpc-ext with the alternative ordering.